packages/fetchai/skills/erc1155_deploy/handlers.py

```diff
@@ -0,0 +1,216 @@
+
+        if fipa_msg.performative == FIPAMessage.Performative.CFP:
+            self._handle_cfp(fipa_msg)
+        elif fipa_msg.performative == FIPAMessage.Performative.INFORM:
+            self._handle_inform(fipa_msg)
+
+    def teardown(self) -> None:
+        """
+        Implement the handler teardown.
+
+        :return: None
+        """
+        pass
+
+    def _handle_cfp(self, msg: FIPAMessage) -> None:
+        """
+        Handle the CFP.
+
+        If the CFP matches the supplied services then send a PROPOSE, otherwise send a DECLINE.
+
+        :param msg: the message
+        :return: None
+        """
+        new_message_id = msg.message_id + 1
+        new_target = msg.message_id
+        self.context.logger.info(
+            "[{}]: received CFP from sender={}".format(
+                self.context.agent_name, msg.counterparty[-5:]
+            )
+        )
+        strategy = cast(Strategy, self.context.strategy)
+        contract = self.context.contracts.erc1155
+        self.context.logger.info(contract.item_ids)
+        contract_nonce = contract.generate_trade_nonce(self.context.agent_address)
+        self.context.shared_state["contract_nonce"] = contract_nonce
+        info_msg = FIPAMessage(
+            message_id=new_message_id,
+            dialogue_reference=("", ""),
+            target=new_target,
+            performative=FIPAMessage.Performative.INFORM,
+            info={
+                "contract": contract.instance.address,
+                "item_ids": contract.item_ids,
+                "trade_nonce": contract_nonce,
+                "from_supply": strategy.from_supply,
+                "to_supply": strategy.to_supply,
+                "value": strategy.value,
+            },
+        )
+
+        self.context.outbox.put_message(
+            to=msg.counterparty,
+            sender=self.context.agent_address,
+            protocol_id=FIPAMessage.protocol_id,
+            message=FIPASerializer().encode(info_msg),
+        )
+
+    def _handle_inform(self, msg: FIPAMessage) -> None:
+        """
+        Handle the INFORM.
+
+        If the INFORM message contains the transaction_digest then verify that it is settled, otherwise do nothing.
+        If the transaction is settled, send the data, otherwise do nothing.
+
+        :param msg: the message
+        :return: None
+        """
+        self.context.logger.info(
+            "[{}]: received INFORM from sender={}".format(
+                self.context.agent_name, msg.counterparty[-5:]
+            )
+        )
+        self.context.logger.info(self.context.ledger_apis.apis.get("ethereum"))
+        strategy = cast(Strategy, self.context.strategy)
+        contract = self.context.contracts.erc1155
+        contract_nonce = cast(int, self.context.shared_state.get("contract_nonce"))
+        tx = contract.get_atomic_swap_single_proposal(
+            from_address=self.context.agent_address,
+            to_address=msg.counterparty,
```

Comment:
I think which  one of the single proposals is used should be configurable.

Reply:
Do you mean that I have to make one function `get_atomic_swap_proporal(proposal_type: int, ....., signature)`
And on the background call the single or batch ?